aws/resource_aws_appautoscaling_scheduled_action.go

```diff
@@ -0,0 +1,222 @@
+				Optional: true,
+				ForceNew: true,
+			},
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppautoscalingScheduledActionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appautoscalingconn
+
+	input := &applicationautoscaling.PutScheduledActionInput{
+		ScheduledActionName: aws.String(d.Get("name").(string)),
+		ServiceNamespace:    aws.String(d.Get("service_namespace").(string)),
+		ResourceId:          aws.String(d.Get("resource_id").(string)),
+	}
+	if v, ok := d.GetOk("scalable_dimension"); ok {
+		input.ScalableDimension = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("schedule"); ok {
+		input.Schedule = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("scalable_target_action"); ok {
+		sta := &applicationautoscaling.ScalableTargetAction{}
+		raw := v.([]interface{})[0].(map[string]interface{})
+		if max, ok := raw["max_capacity"]; ok {
+			sta.MaxCapacity = aws.Int64(int64(max.(int)))
+		}
+		if min, ok := raw["min_capacity"]; ok {
+			sta.MaxCapacity = aws.Int64(int64(min.(int)))
+		}
+		input.ScalableTargetAction = sta
+	}
+	if v, ok := d.GetOk("start_time"); ok {
+		t, err := time.Parse(awsAppautoscalingScheduleTimeLayout, v.(string))
+		if err != nil {
+			return fmt.Errorf("Error Parsing Appautoscaling Scheduled Action Start Time: %s", err.Error())
+		}
+		input.StartTime = aws.Time(t)
+	}
+	if v, ok := d.GetOk("end_time"); ok {
+		t, err := time.Parse(awsAppautoscalingScheduleTimeLayout, v.(string))
+		if err != nil {
+			return fmt.Errorf("Error Parsing Appautoscaling Scheduled Action End Time: %s", err.Error())
+		}
+		input.EndTime = aws.Time(t)
+	}
+
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.PutScheduledAction(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case applicationautoscaling.ErrCodeObjectNotFoundException:
+					return resource.RetryableError(err)
+				default:
+					return resource.NonRetryableError(err)
+				}
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("name").(string))
+	return resourceAwsAppautoscalingScheduledActionRead(d, meta)
+}
+
+func resourceAwsAppautoscalingScheduledActionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appautoscalingconn
+
+	saName := d.Id()
+	input := &applicationautoscaling.DescribeScheduledActionsInput{
+		ScheduledActionNames: []*string{aws.String(saName)},
+		ServiceNamespace:     aws.String(d.Get("service_namespace").(string)),
+	}
+	resp, err := conn.DescribeScheduledActions(input)
+	if err != nil {
+		return err
+	}
+	if len(resp.ScheduledActions) < 1 {
+		d.SetId("")
+		return nil
+	}
+	if len(resp.ScheduledActions) != 1 {
+		return fmt.Errorf("Number of Scheduled Action (%s) isn't one, got %d", saName, len(resp.ScheduledActions))
```

Comment:
Grammar nitpick, do you mind changing this to `Expected 1 scheduled action under %s, found %d`?